src/main/scala/org/broadinstitute/dsde/firecloud/service/RegisterService.scala

```diff
@@ -56,7 +57,15 @@ class RegisterService(val rawlsDao: RawlsDAO, val samDao: SamDAO, val thurloeDao
         for {
           registrationInfo <- samDao.registerUser(userInfo)
           _ <- rawlsDao.registerUser(userInfo) //This call to rawls handles leftover registration pieces (welcome email and pending workspace access)
-        } yield registrationInfo
+          freeCredits:Either[Exception,UserTrialStatus] <- enableSelfForFreeCredits(userInfo)
+            .map(Right(_)) recover { case e: Exception => Left(e) }
+        } yield {
+          val message = freeCredits match {
+            case Left(ex) => Some(s"Error enabling free credits during registration. Underlying error: ${ex.getMessage}")
+            case Right(_) => None
+          }
+          registrationInfo.copy(message = message)
```

Comment:
Perhaps we should append the free credits `message` to the existing message, instead of replacing it.

Reply:
Good call; I'll change. I know this is safe for now because I just added the `message` field and therefore know nothing else writes to it ... but it's worth changing for the future.